injected/integration-test/web-compat-android.spec.js

```diff
@@ -0,0 +1,270 @@
+                    await navigate(page)
+                    await beforeEach(page)
+                    const { result, message } = await checkShare(page, { text: 'xxx', url: 'http://example.com' })
+                    expect(message).toMatchObject({ featureName: 'webCompat', method: 'webShare', params: { text: 'xxx http://example.com/' } })
+                    expect(result).toBeUndefined()
+                })
+
+                test('should throw when sharing files', async ({ page }) => {
+                    await navigate(page)
+                    await beforeEach(page)
+                    const { result, message } = await checkShare(page, { title: 'title', files: [] })
+                    expect(message).toBeNull()
+                    expect(result.threw.message).toContain('TypeError: Invalid share data')
+                })
+
+                test('should throw when sharing non-http urls', async ({ page }) => {
+                    await navigate(page)
+                    await beforeEach(page)
+                    const { result, message } = await checkShare(page, { url: 'chrome://bla' })
+                    expect(message).toBeNull()
+                    expect(result.threw.message).toContain('TypeError: Invalid share data')
+                })
+
+                test('should handle relative urls', async ({ page }) => {
+                    await navigate(page)
+                    await beforeEach(page)
+                    const { result, message } = await checkShare(page, { url: 'bla' })
+                    expect(message.params.url).toMatch(/^http:\/\/localhost:\d+\/bla$/)
+                    expect(result).toBeUndefined()
+                })
+
+                test('should treat empty url as relative', async ({ page }) => {
+                    await navigate(page)
+                    await beforeEach(page)
+                    const { result, message } = await checkShare(page, { url: '' })
+                    expect(message.params.url).toMatch(/^http:\/\/localhost:\d+\//)
+                    expect(result).toBeUndefined()
+                })
+            })
+
+            test.describe('(handling errors from Android)', () => {
+                test('should handle messaging error', async ({ page }) => {
+                    // page.on('console', (msg) => console.log(msg.type(), msg.text()))
+                    await navigate(page)
+                    await beforeEach(page)
+
+                    await page.evaluate(() => {
+                        globalThis.cssMessaging.impl.request = () => {
+                            return Promise.reject(new Error('something wrong'))
+                        }
+                    })
+                    const result = await page.evaluate('navigator.share({ text: "xxx" })').catch((e) => {
+                        return { threw: e }
+                    })
+
+                    expect(result.threw.message).toContain('DataError: something wrong')
```

Comment:
This is the difference, before you were expecting DOMException - but I tested in modern browsers and I don't think that's correct. If you throw a new DOMException("oops", "ABC") - then the expected message should be "ABC: oops", and *NOT* include DOMException.

I may have have mis-understood here, so let me know. but I think the updated browsers in playwright have exposed this :)